src/shogun/machine/Machine.h

```diff
@@ -255,6 +274,21 @@ class Machine : public StoppableSGObject
 			return false;
 		}
 
+		virtual std::shared_ptr<Machine> fit_machine(
+		    const std::shared_ptr<Features>& data,
+		    const std::shared_ptr<Labels>& lab)
+		{
+			error("train_machine is not yet implemented for {}!", get_name());
+			return nullptr;
+		}
```

Comment:
I think this should be pure virtual no? I guess for now keep it, because otherwise it will be too much work to refactor everything in one go

Reply:
++